### DISCUSSION TITLE
Multiple Output for dicom series to volume operator

### DISCUSSION THREAD

**Discussion by vikashg** [Project-MONAI/monai-deploy-app-sdk#342] [Q&A]
Hi everyone, 
Any application follows the usual path `dicom_data_loader_op` -> `series_selector_op` -> `series_to_vol_op`. If an empty string is passed on to the series_selector_op, it selects all the series coming in from the previous operator. However, the series to volume operator returns only a single volume as is seen [here](https://github.com/Project-MONAI/monai-deploy-app-sdk/blob/main/monai/deploy/operators/dicom_series_to_volume_operator.py) 
```
        # TODO: This needs to be updated once allowed to output multiple Image objects
        return study_selected_series_list[0].selected_series[0].image
```
So, if we have to do inference on every single series , may be we need to pass each series independently. 
Am i correct in thinking so ?

Thanks 
Vikash

**Comment by MMelQin** [comment]
Good question @vikashg! Yes, you have answered your own question. Let's look at this and a more advanced use case for more details.

1. AI model takes in a single image (3D or 2D), each coming from a converted DICOM series, but you have a batch of DICOM series all needing to be process independently.

- Construct the series selection rule so that specific series can be selected, just in case multiple series in the same or different study are used as the input to the application
- Use the batch processing logic in the app's main function to iterate through the stack of DICOM study/series in the file system, so that each invocation of the app's `run` method will result in a single series/image. Please see Discussion topic #333 , and please see PR #337 for code examples.

2. More advanced use case. Two or more images are needed, each from a DICOM series, and the images either need to be registered into the same image or be used as multiple inputs to a model

- The selection rules will need to contain multiple named selections, each will result in a selected series indexed by the name respectively.
- Consuming the output from the series selector will require a custom operator or more, each will parse and select a name indexed series and then call the series to vol to convert to image (note here, the series to vol will get only one series), whose output will then be fed into the model, linked to the specific named input.
- The technique described here has practical relevance for multiple input model(s), e.g. Oncology Board use case where both prior(s) and current series will be analyzed to detect/classify the progression of the treatment. The inference operator is expected to have multiple named inputs, each expecting a converted image from a series (current and specific prior(s)).

**Reply by vikashg** [reply]
Thanks @MMelQin Yes I will look into this STL Operator code to get some insights. Just when the connectathon gets over :)
